ega-data-api-htsget/src/test/java/eu/elixir/ega/ebi/htsget/rest/V2IntegrationTest.java

```diff
@@ -0,0 +1,101 @@
+package eu.elixir.ega.ebi.htsget.rest;
+
+import eu.elixir.ega.ebi.commons.shared.config.NotFoundException;
+import eu.elixir.ega.ebi.commons.shared.dto.File;
+import eu.elixir.ega.ebi.commons.shared.dto.FileIndexFile;
+import eu.elixir.ega.ebi.commons.shared.dto.MyExternalConfig;
+import eu.elixir.ega.ebi.commons.shared.service.FileInfoService;
+import eu.elixir.ega.ebi.htsget.HtsgetServiceApplication;
+import eu.elixir.ega.ebi.htsget.config.LocalTestData;
+import eu.elixir.ega.ebi.htsget.formats.DataProviderFactory;
+import eu.elixir.ega.ebi.htsget.service.TicketServiceV2;
+import eu.elixir.ega.ebi.htsget.service.internal.ResClient;
+import eu.elixir.ega.ebi.htsget.service.internal.TicketServiceV2Impl;
+import htsjdk.samtools.seekablestream.SeekableFileStream;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureWebMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.FileNotFoundException;
+import java.net.MalformedURLException;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doReturn;
+
+@RunWith(SpringRunner.class)
+@AutoConfigureWebMvc
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT, classes = {HtsgetServiceApplication.class})
+@TestPropertySource(locations = "classpath:application-test.properties")
+public class V2IntegrationTest {
+
+    @Autowired
+    public TicketControllerV2 controller;
+
+    @Autowired
+    TestRestTemplate restTemplate;
+
+    @Test
+    @Ignore
+    public void runService() {
+
+        String versionStr = restTemplate.getForObject("/htsget/version", String.class);
+        Assert.assertEquals("v1.0.0", versionStr);
+
+        while (true) {
+
+        }
+    }
+
+    @TestConfiguration
```

Comment:
Prefer to put the test configuration in their own classes and import them.

Reply:
I can do that, but could you help me understand why it would be better? I thought this way would be better because it makes it clear that this configuration is only for the tests in this file, and it means that the configuration and tests are kept together.

Reply:
It's more a rule of thumb to try to leave the code that is test test separated from the configuration. it is not a hard rule. Usually what I do is have [whatever]test.java and [whatever]testConfiguration.java